service/history/handler.go

```diff
@@ -1753,6 +1756,50 @@ func (h *handlerImpl) ReapplyEvents(
 	return nil
 }
 
+func (h *handlerImpl) CountDLQMessages(
+	ctx context.Context,
+	request *types.CountDLQMessagesRequest,
+) (resp *types.HistoryCountDLQMessagesResponse, retError error) {
+	defer log.CapturePanic(h.GetLogger(), &retError)
+	h.startWG.Wait()
+
+	scope, sw := h.startRequestProfile(ctx, metrics.HistoryCountDLQMessagesScope)
+	defer sw.Stop()
+
+	if h.isShuttingDown() {
+		return nil, errShuttingDown
+	}
+
+	g := &errgroup.Group{}
+	var mu sync.Mutex
+	entries := map[types.HistoryDLQCountKey]int64{}
+	for _, shardID := range h.controller.ShardIDs() {
+		shardID := shardID
+		g.Go(func() error {
```

Comment:
Should the panic handler not be in here?

Reply:
This and also few other places did not have it. But yeah, it is probably a good idea.
Added log.CapturePanic for this and other missing places. It logs the panic and returns it as error.